# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/helix/consistent-tab-width.patch
# Copyright (C) 2024 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

Helix's default embedded non-modifiable tree-sitter config sets tab-width to
various value (2, 3, 4, 8). Make it more consistent by setting it to 4 so that
visual do not change between files that have different LSPs and/or tree-sitter
grammar attached to them.

  - NoTag <notag@t2sde.org>

--- helix-24.07/languages.toml.vanilla	2024-10-14 10:29:04.768993156 +0200
+++ helix-24.07/languages.toml	2024-10-14 10:30:51.773014787 +0200
@@ -300,7 +300,7 @@
 file-types = ["toml", { glob = "pdm.lock" }, { glob = "poetry.lock" }, { glob = "Cargo.lock" }, { glob = "uv.lock" }]
 comment-token = "#"
 language-servers = [ "taplo" ]
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [[grammar]]
 name = "toml"
@@ -313,7 +313,7 @@
 file-types = ["awk", "gawk", "nawk", "mawk"]
 comment-token = "#"
 language-servers = [ "awk-language-server" ]
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [[grammar]]
 name = "awk"
@@ -327,7 +327,7 @@
 language-servers = [ "bufls", "pbkit" ]
 comment-token = "//"
 block-comment-tokens = { start = "/*", end = "*/" }
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [[grammar]]
 name = "protobuf"
@@ -342,7 +342,7 @@
 roots = ["mix.exs", "mix.lock"]
 comment-token = "#"
 language-servers = [ "elixir-ls" ]
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [[grammar]]
 name = "elixir"
@@ -372,7 +372,7 @@
 comment-token = "//"
 block-comment-tokens = { start = "/*", end = "*/" }
 language-servers = [ "mint" ]
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [[language]]
 name = "mojo"
@@ -398,7 +398,7 @@
 shebangs = ["janet"]
 roots = ["project.janet"]
 comment-token = "#"
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 formatter = { command = "janet-format" }
 grammar = "janet-simple"
 
@@ -442,7 +442,7 @@
 ]
 language-servers = [ "vscode-json-language-server" ]
 auto-format = true
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [[grammar]]
 name = "json"
@@ -456,7 +456,7 @@
 grammar = "json"
 language-servers = [ "vscode-json-language-server" ]
 auto-format = true
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [[language]]
 name = "json5"
@@ -480,7 +480,7 @@
 comment-token = "//"
 block-comment-tokens = { start = "/*", end = "*/" }
 language-servers = [ "clangd" ]
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [language.debugger]
 name = "lldb-dap"
@@ -517,7 +517,7 @@
 comment-token = "//"
 block-comment-tokens = { start = "/*", end = "*/" }
 language-servers = [ "clangd" ]
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [language.debugger]
 name = "lldb-dap"
@@ -552,7 +552,7 @@
 file-types = ["cr"]
 roots = ["shard.yml", "shard.lock"]
 comment-token = "#"
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 grammar = "ruby"
 language-servers = [ "crystalline" ]
 
@@ -596,7 +596,7 @@
 injection-regex = "cel"
 file-types = ["cel"]
 comment-token = "//"
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [[grammar]]
 name = "cel"
@@ -608,7 +608,7 @@
 injection-regex = "spicedb"
 file-types = ["zed"]
 comment-token = "//"
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [[grammar]]
 name = "spicedb"
@@ -690,7 +690,7 @@
 comment-token = "//"
 block-comment-tokens = { start = "/*", end = "*/" }
 language-servers = [ "gopls" ]
-indent = { tab-width = 2, unit = " " }
+indent = { tab-width = 4, unit = " " }
 
 [[grammar]]
 name = "gotmpl"
@@ -720,7 +720,7 @@
 comment-token = "//"
 block-comment-tokens = { start = "/*", end = "*/" }
 language-servers = [ "typescript-language-server" ]
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [language.debugger]
 name = "node-debug2"
@@ -747,7 +747,7 @@
 comment-token = "//"
 block-comment-tokens = { start = "/*", end = "*/" }
 language-servers = [ "typescript-language-server" ]
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 grammar = "javascript"
 
 [[language]]
@@ -760,7 +760,7 @@
 comment-token = "//"
 block-comment-tokens = { start = "/*", end = "*/" }
 language-servers = [ "typescript-language-server" ]
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [[grammar]]
 name = "typescript"
@@ -775,7 +775,7 @@
 comment-token = "//"
 block-comment-tokens = { start = "/*", end = "*/" }
 language-servers = [ "typescript-language-server" ]
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [[grammar]]
 name = "tsx"
@@ -789,7 +789,7 @@
 block-comment-tokens = { start = "/*", end = "*/" }
 language-servers = [ "vscode-css-language-server" ]
 auto-format = true
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [[grammar]]
 name = "css"
@@ -803,7 +803,7 @@
 block-comment-tokens = { start = "/*", end = "*/" }
 language-servers = [ "vscode-css-language-server" ]
 auto-format = true
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [[grammar]]
 name = "scss"
@@ -817,7 +817,7 @@
 block-comment-tokens = { start = "<!--", end = "-->" }
 language-servers = [ "vscode-html-language-server" ]
 auto-format = true
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [[grammar]]
 name = "html"
@@ -847,7 +847,7 @@
 shebangs = []
 comment-token = "#"
 language-servers = [ "nls" ]
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [language.auto-pairs]
 '(' = ')'
@@ -867,7 +867,7 @@
 shebangs = []
 comment-token = "#"
 language-servers = [ "nil" ]
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [[grammar]]
 name = "nix"
@@ -916,7 +916,7 @@
 shebangs = ["ruby"]
 comment-token = "#"
 language-servers = [ "solargraph" ]
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [[grammar]]
 name = "ruby"
@@ -974,7 +974,7 @@
 shebangs = ["sh", "bash", "dash", "zsh"]
 comment-token = "#"
 language-servers = [ "bash-language-server" ]
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [[grammar]]
 name = "bash"
@@ -1024,7 +1024,7 @@
 injection-regex = "twig"
 file-types = ["twig"]
 block-comment-tokens = { start = "{#", end = "#}" }
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [[grammar]]
 name = "twig"
@@ -1079,7 +1079,7 @@
 comment-token = "--"
 block-comment-tokens = { start = "/-", end = "-/" }
 language-servers = [ "lean" ]
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 rulers = [101]
 text-width = 100
 
@@ -1128,7 +1128,7 @@
 file-types = ["java", "jav", "pde"]
 roots = ["pom.xml", "build.gradle", "build.gradle.kts"]
 language-servers = [ "jdtls" ]
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [[grammar]]
 name = "java"
@@ -1165,7 +1165,7 @@
 injection-regex = "beancount"
 file-types = ["beancount", "bean"]
 comment-token = ";"
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [[grammar]]
 name = "beancount"
@@ -1179,7 +1179,7 @@
 shebangs = ["ocaml", "ocamlrun", "ocamlscript"]
 block-comment-tokens = { start = "(*", end = "*)" }
 language-servers = [ "ocamllsp" ]
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [language.auto-pairs]
 '(' = ')'
@@ -1199,7 +1199,7 @@
 block-comment-tokens = { start = "(*", end = "*)" }
 comment-token = "(**)"
 language-servers = [ "ocamllsp" ]
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [language.auto-pairs]
 '(' = ')'
@@ -1220,7 +1220,7 @@
 roots = [".luarc.json", ".luacheckrc", ".stylua.toml", "selene.toml", ".git"]
 comment-token = "--"
 block-comment-tokens = { start = "--[[", end = "--]]" }
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 language-servers = [ "lua-language-server" ]
 
 [[grammar]]
@@ -1232,7 +1232,7 @@
 scope = "source.svelte"
 injection-regex = "svelte"
 file-types = ["svelte"]
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 language-servers = [ "svelteserver" ]
 
 [[grammar]]
@@ -1246,7 +1246,7 @@
 file-types = ["vue"]
 roots = ["package.json"]
 block-comment-tokens = { start = "<!--", end = "-->" }
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 language-servers = [ "vuels" ]
 
 [[grammar]]
@@ -1258,7 +1258,7 @@
 scope = "source.yaml"
 file-types = ["yml", "yaml"]
 comment-token = "#"
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 language-servers = [ "yaml-language-server", "ansible-language-server" ]
 injection-regex = "yml|yaml"
 
@@ -1275,7 +1275,7 @@
 comment-token = "--"
 block-comment-tokens = { start = "{-", end = "-}" }
 language-servers = [ "haskell-language-server" ]
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [[grammar]]
 name = "haskell"
@@ -1286,7 +1286,7 @@
 scope = "source.persistentmodels"
 file-types = ["persistentmodels"]
 comment-token = "--"
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [[grammar]]
 name = "haskell-persistent"
@@ -1301,7 +1301,7 @@
 comment-token = "--"
 block-comment-tokens = { start = "{-", end = "-}" }
 language-servers = [ "purescript-language-server" ]
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 auto-format = true
 formatter = { command = "purs-tidy", args = ["format"] }
 
@@ -1363,7 +1363,7 @@
 file-types = ["tsq"]
 comment-token = ";"
 injection-regex = "tsq"
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [[grammar]]
 name = "tsq"
@@ -1375,7 +1375,7 @@
 file-types = ["cmake", { glob = "CMakeLists.txt" }]
 comment-token = "#"
 block-comment-tokens = { start = "#[[", end = "]]" }
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 language-servers = [ "cmake-language-server" ]
 injection-regex = "cmake"
 
@@ -1416,7 +1416,7 @@
 shebangs = ["perl"]
 comment-token = "#"
 language-servers = [ "perlnavigator" ]
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [[grammar]]
 name = "perl"
@@ -1438,7 +1438,7 @@
 file-types = ["rkt", "rktd", "rktl", "scrbl"]
 shebangs = ["racket"]
 comment-token = ";"
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 block-comment-tokens = { start = "#|", end = "|#" }
 language-servers = [ "racket" ]
 grammar = "scheme"
@@ -1449,7 +1449,7 @@
 file-types = ["lisp", "asd", "cl", "l", "lsp", "ny", "podsl", "sexp"]
 shebangs = ["lisp", "sbcl", "ccl", "clisp", "ecl"]
 comment-token = ";"
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 language-servers = [ "cl-lsp" ]
 grammar = "scheme"
 
@@ -1487,7 +1487,7 @@
 scope = "source.llvm"
 file-types = ["ll"]
 comment-token = ";"
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 injection-regex = "llvm"
 
 [[grammar]]
@@ -1499,7 +1499,7 @@
 scope = "source.llvm_mir"
 file-types = []
 comment-token = ";"
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 injection-regex = "mir"
 
 [[grammar]]
@@ -1514,7 +1514,7 @@
 scope = "source.yaml"
 file-types = ["mir"]
 comment-token = "#"
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [[language]]
 name = "tablegen"
@@ -1522,7 +1522,7 @@
 file-types = ["td"]
 comment-token = "//"
 block-comment-tokens = { start = "/*", end = "*/" }
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 injection-regex = "tablegen"
 
 [[grammar]]
@@ -1536,7 +1536,7 @@
 file-types = ["md", "markdown", "mkd", "mkdn", "mdwn", "mdown", "markdn", "mdtxt", "mdtext", "workbook", { glob = "PULLREQ_EDITMSG" }]
 roots = [".marksman.toml"]
 language-servers = [ "marksman", "markdown-oxide" ]
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 block-comment-tokens = { start = "<!--", end = "-->" }
 
 [[grammar]]
@@ -1563,7 +1563,7 @@
 comment-token = "//"
 block-comment-tokens = { start = "/*", end = "*/" }
 language-servers = [ "dart" ]
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [[grammar]]
 name = "dart"
@@ -1576,7 +1576,7 @@
 file-types = ["scala", "sbt", "sc"]
 comment-token = "//"
 block-comment-tokens = { start = "/*", end = "*/" }
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 language-servers = [ "metals" ]
 
 [[grammar]]
@@ -1603,7 +1603,7 @@
   { glob = "containerfile.*" },
 ]
 comment-token = "#"
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 language-servers = [ "docker-langserver" ]
 
 [[grammar]]
@@ -1617,7 +1617,7 @@
 language-servers = [ "docker-compose-langserver", "yaml-language-server" ]
 file-types = [{ glob = "docker-compose.yaml" }, { glob = "docker-compose.yml" }]
 comment-token = "#"
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 grammar = "yaml"
 
 [[language]]
@@ -1625,7 +1625,7 @@
 scope = "git.commitmsg"
 file-types = [{ glob = "COMMIT_EDITMSG" }]
 comment-token = "#"
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 rulers = [51, 73]
 text-width = 72
 
@@ -1639,7 +1639,7 @@
 file-types = ["diff", "patch", "rej"]
 injection-regex = "diff"
 comment-token = "#"
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [[grammar]]
 name = "diff"
@@ -1651,7 +1651,7 @@
 file-types = [{ glob = "git-rebase-todo" }]
 injection-regex = "git-rebase"
 comment-token = "#"
-indent = { tab-width = 2, unit = "y" }
+indent = { tab-width = 4, unit = "y" }
 
 [[grammar]]
 name = "git-rebase"
@@ -1711,7 +1711,7 @@
 language-servers = [ "graphql-language-service" ]
 comment-token = "#"
 block-comment-tokens = { start = "\"\"\"", end = "\"\"\"" }
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [[grammar]]
 name = "graphql"
@@ -1754,7 +1754,7 @@
 comment-token = "//"
 block-comment-tokens = { start = "/*", end = "*/" }
 language-servers = [ "rescript-language-server" ]
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [[grammar]]
 name = "rescript"
@@ -1797,7 +1797,7 @@
 file-types = ["hcl", "tf", "nomad"]
 comment-token = "#"
 block-comment-tokens = { start = "/*", end = "*/" }
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 language-servers = [ "terraform-ls" ]
 auto-format = true
 
@@ -1812,7 +1812,7 @@
 file-types = ["tfvars"]
 comment-token = "#"
 block-comment-tokens = { start = "/*", end = "*/" }
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 language-servers = [ "terraform-ls" ]
 auto-format = true
 grammar = "hcl"
@@ -1822,7 +1822,7 @@
 scope = "source.org"
 injection-regex = "org"
 file-types = ["org"]
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [[grammar]]
 name = "org"
@@ -1849,7 +1849,7 @@
 file-types = ["gleam"]
 roots = ["gleam.toml"]
 comment-token = "//"
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 language-servers = [ "gleam" ]
 auto-format = true
 
@@ -1890,7 +1890,7 @@
 file-types = ["r", "R", { glob = ".Rprofile" }, { glob = "Rprofile.site" }, { glob = ".RHistory" }]
 shebangs = ["r", "R"]
 comment-token = "#"
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 language-servers = [ "r" ]
 
 [[grammar]]
@@ -1903,7 +1903,7 @@
 language-id = "rmd"
 injection-regex = "(r|R)md"
 file-types = ["rmd", "Rmd"]
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 grammar = "markdown"
 block-comment-tokens = { start = "<!--", end = "-->" }
 language-servers = [ "r" ]
@@ -1930,7 +1930,7 @@
 injection-regex = "erb"
 file-types = ["erb"]
 block-comment-tokens = { start = "<!--", end = "-->" }
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 grammar = "embedded-template"
 
 [[language]]
@@ -1939,7 +1939,7 @@
 injection-regex = "ejs"
 file-types = ["ejs"]
 block-comment-tokens = { start = "<!--", end = "-->" }
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 grammar = "embedded-template"
 
 [[grammar]]
@@ -1953,7 +1953,7 @@
 file-types = ["eex"]
 roots = ["mix.exs", "mix.lock"]
 block-comment-tokens = { start = "<!--", end = "-->" }
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [[grammar]]
 name = "eex"
@@ -1966,7 +1966,7 @@
 file-types = ["heex"]
 roots = ["mix.exs", "mix.lock"]
 block-comment-tokens = { start = "<!--", end = "-->" }
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 language-servers = [ "elixir-ls" ]
 
 [[grammar]]
@@ -2024,7 +2024,7 @@
 file-types = ["nu", "nuon"]
 shebangs = ["nu"]
 comment-token = "#"
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 language-servers = [ "nu-lsp" ]
 
 [[grammar]]
@@ -2038,7 +2038,7 @@
 file-types = ["vala", "vapi"]
 comment-token = "//"
 block-comment-tokens = { start = "/*", end = "*/" }
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 language-servers = [ "vala-language-server" ]
 
 [[grammar]]
@@ -2051,7 +2051,7 @@
 injection-regex = "hare"
 file-types = ["ha"]
 comment-token = "//"
-indent = { tab-width = 8, unit = "\t" }
+indent = { tab-width = 4, unit = "\t" }
 
 [[grammar]]
 name = "hare"
@@ -2091,7 +2091,7 @@
 file-types = ["cpon", "cp"]
 auto-format = true
 comment-token = "//"
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [[grammar]]
 name = "cpon"
@@ -2142,7 +2142,7 @@
 injection-regex = "meson"
 file-types = [{ glob = "meson.build" }, { glob = "meson.options" }, { glob = "meson_options.txt" }]
 comment-token = "#"
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [[grammar]]
 name = "meson"
@@ -2165,7 +2165,7 @@
 file-types = ["ss", "scm"]
 shebangs = ["scheme", "guile", "chicken"]
 comment-token = ";"
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [language.auto-pairs]
 '(' = ')'
@@ -2200,7 +2200,7 @@
 comment-token = "//"
 block-comment-tokens = { start = "/*", end = "*/" }
 language-servers = [ "svlangserver" ]
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 injection-regex = "verilog"
 
 [[grammar]]
@@ -2223,7 +2223,7 @@
 scope = "source.jsdoc"
 injection-regex = "jsdoc"
 file-types = ["jsdoc"]
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [[grammar]]
 name = "jsdoc"
@@ -2237,7 +2237,7 @@
 comment-token = "//"
 block-comment-tokens = { start = "/*", end = "*/" }
 language-servers = [ "openscad-lsp" ]
-indent = { tab-width = 2, unit = "\t" }
+indent = { tab-width = 4, unit = "\t" }
 
 [[grammar]]
 name = "openscad"
@@ -2251,7 +2251,7 @@
 roots = ["package.json"]
 comment-token = "//"
 language-servers = [ "prisma-language-server" ]
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [[grammar]]
 name = "prisma"
@@ -2265,7 +2265,7 @@
 roots = ["project.clj", "build.boot", "deps.edn", "shadow-cljs.edn"]
 comment-token = ";"
 language-servers = [ "clojure-lsp" ]
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [[grammar]]
 name = "clojure"
@@ -2286,7 +2286,7 @@
 shebangs = ["elvish"]
 file-types = ["elv"]
 comment-token = "#"
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 language-servers = [ "elvish" ]
 grammar = "elvish"
 
@@ -2302,7 +2302,7 @@
 shebangs = []
 comment-token = "--"
 block-comment-tokens = { start = "{-", end = "-}" }
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 language-servers = [ "idris2-lsp" ]
 
 [[language]]
@@ -2325,7 +2325,7 @@
 injection-regex = "ungrammar"
 file-types = ["ungram", "ungrammar"]
 comment-token = "//"
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [[grammar]]
 name = "ungrammar"
@@ -2381,7 +2381,7 @@
 injection-regex = "task"
 file-types = ["task"]
 comment-token = "#"
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [[grammar]]
 name = "task"
@@ -2404,7 +2404,7 @@
 injection-regex = "esdl"
 file-types = ["esdl"]
 comment-token = "#"
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 roots = ["edgedb.toml"]
 
 [[grammar]]
@@ -2418,7 +2418,7 @@
 file-types = ["pas", "pp", "inc", "lpr", "lfm"]
 comment-token = "//"
 block-comment-tokens = { start = "{", end = "}" }
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 language-servers = [ "pasls" ]
 
 [[grammar]]
@@ -2443,7 +2443,7 @@
 roots = ["jsonnetfile.json"]
 comment-token = "//"
 block-comment-tokens = { start = "/*", end = "*/" }
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 language-servers = [ "jsonnet-language-server" ]
 
 [[grammar]]
@@ -2457,7 +2457,7 @@
 file-types = ["adb", "ads", "gpr"]
 roots = ["alire.toml"]
 comment-token = "--"
-indent = { tab-width = 3, unit = "   " }
+indent = { tab-width = 4, unit = "   " }
 language-servers = ["ada-language-server", "ada-gpr-language-server"]
 
 
@@ -2471,7 +2471,7 @@
 injection-regex = "astro"
 file-types = ["astro"]
 block-comment-tokens = { start = "<!--", end = "-->" }
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [[grammar]]
 name = "astro"
@@ -2483,7 +2483,7 @@
 injection-regex = "bass"
 file-types = ["bass"]
 comment-token = ";"
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 language-servers = [ "bass" ]
 
 [[grammar]]
@@ -2532,7 +2532,7 @@
 scope = "source.vhs"
 file-types = ["tape"]
 comment-token = "#"
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 grammar = "vhs"
 
 [[grammar]]
@@ -2617,7 +2617,7 @@
   "ui"
 ]
 block-comment-tokens = { start = "<!--", end = "-->" }
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [language.auto-pairs]
 '(' = ')'
@@ -2637,7 +2637,7 @@
 scope = "source.dtd"
 injection-regex = "dtd"
 file-types = ["dtd", "ent"]
-indent = {tab-width = 2, unit = "  "}
+indent = {tab-width = 4, unit = "  "}
 
 [language.auto-pairs]
 '(' = ')'
@@ -2657,7 +2657,7 @@
 file-types = ["wit"]
 comment-token = "//"
 block-comment-tokens = { start = "/*", end = "*/" }
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [language.auto-pairs]
 '(' = ')'
@@ -2726,7 +2726,7 @@
 file-types = ["inko"]
 roots = ["inko.pkg"]
 comment-token = "#"
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 formatter = { command = "inko", args = ["fmt", "-"] }
 
 [[grammar]]
@@ -2740,7 +2740,7 @@
 auto-format = true
 comment-token = "//"
 block-comment-tokens = { start = "/*", end = "*/" }
-indent = { tab-width = 2, unit = " "}
+indent = { tab-width = 4, unit = " "}
 language-servers = [ "bicep-langserver" ]
 
 [[grammar]]
@@ -2779,7 +2779,7 @@
 file-types = ["m"]
 comment-token = "%"
 shebangs = ["octave-cli", "matlab"]
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [[grammar]]
 name = "matlab"
@@ -2791,7 +2791,7 @@
 file-types = ["pony"]
 injection-regex = "pony"
 roots = ["corral.json", "lock.json"]
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 comment-token = "//"
 block-comment-tokens = { start = "/*", end = "*/" }
 
@@ -2806,7 +2806,7 @@
 file-types = ["dhall"]
 comment-token = "--"
 block-comment-tokens = { start = "{-", end = "-}" }
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 language-servers = [ "dhall-lsp-server" ]
 formatter = { command = "dhall" , args = ["format"] }
 
@@ -2828,7 +2828,7 @@
 scope = "source.msbuild"
 injection-regex = "msbuild"
 file-types = ["proj", "vbproj", "csproj", "fsproj", "targets", "props"]
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 block-comment-tokens = { start = "<!--", end = "-->" }
 grammar = "xml"
 
@@ -2888,7 +2888,7 @@
 injection-regex = "yuck"
 file-types = ["yuck"]
 comment-token = ";"
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [[grammar]]
 name = "yuck"
@@ -2922,7 +2922,7 @@
 file-types = ["asm", "S", "nasm"]
 injection-regex = "n?asm"
 comment-token = ";"
-indent = { tab-width = 8, unit = "        " }
+indent = { tab-width = 4, unit = "        " }
 
 [[grammar]]
 name = "nasm"
@@ -2934,7 +2934,7 @@
 file-types = ["s"]
 injection-regex = "gas"
 comment-token = "#"
-indent = { tab-width = 8, unit = "        " }
+indent = { tab-width = 4, unit = "        " }
 
 [[grammar]]
 name = "gas"
@@ -2956,7 +2956,7 @@
 injection-regex = "capnp"
 file-types = ["capnp"]
 comment-token = "#"
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [[grammar]]
 name = "capnp"
@@ -2982,7 +2982,7 @@
 file-types = ["vhd", "vhdl"]
 comment-token = "--"
 language-servers = [ "vhdl_ls" ]
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 injection-regex = "vhdl"
 
 [[grammar]]
@@ -3011,7 +3011,7 @@
 shebangs = []
 comment-token = "#"
 block-comment-tokens = { start = "#[", end = "]#" }
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 language-servers = [ "nimlangserver" ]
 
 [language.auto-pairs]
@@ -3030,7 +3030,7 @@
 scope = "source.cabal"
 file-types = [ "cabal" ]
 roots = ["cabal.project", "Setup.hs"]
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 comment-token = "--"
 language-servers = [ "haskell-language-server" ]
 
@@ -3041,7 +3041,7 @@
 file-types = ["hurl"]
 comment-token = "#"
 formatter = { command = "hurlfmt" }
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [[grammar]]
 name = "hurl"
@@ -3091,7 +3091,7 @@
 file-types = ["gn", "gni"]
 roots = []
 comment-token = "#"
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 formatter = { command = "gn", args = ["format", "--stdin"] }
 
 [[grammar]]
@@ -3119,7 +3119,7 @@
 file-types = ["fs", "forth", "fth", "4th"]
 comment-token = "\\"
 language-servers = [ "forth-lsp" ]
-indent = { tab-width = 3, unit = "   " }
+indent = { tab-width = 4, unit = "   " }
 
 [[grammar]]
 name = "forth"
@@ -3147,7 +3147,7 @@
 injection-regex = "t32"
 file-types = ["cmm", "t32"]
 comment-token = ";"
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [[grammar]]
 name = "t32"
@@ -3159,7 +3159,7 @@
 injection-regex = "webc"
 file-types = ["webc"]
 block-comment-tokens = { start = "<!--", end = "-->" }
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 grammar = "html"
 
 [[language]]
@@ -3170,7 +3170,7 @@
 comment-token = "//"
 block-comment-tokens = { start = "/*", end = "*/" }
 language-servers = ["tinymist", "typst-lsp"]
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [language.auto-pairs]
 '(' = ')'
@@ -3188,7 +3188,7 @@
 scope = "text.html.nunjucks"
 injection-regex = "nunjucks"
 file-types = ["njk"]
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 grammar = "jinja2"
 
 [[language]]
@@ -3196,7 +3196,7 @@
 scope = "text.html.jinja"
 injection-regex = "jinja"
 file-types = ["jinja", "jinja2", "j2"]
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 grammar = "jinja2"
 
 [[grammar]]
@@ -3212,7 +3212,7 @@
 scope = "source.wren"
 injection-regex = "wren"
 file-types = ["wren"]
-indent = { tab-width = 2, unit = "  "}
+indent = { tab-width = 4, unit = "  "}
 
 [[language]]
 name = "unison"
@@ -3275,7 +3275,7 @@
 # language-servers = [ "als" ]
 # the agda language server is of questionable functionality.
 auto-format = false
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [language.auto-pairs]
 '"' = '"'
@@ -3297,7 +3297,7 @@
 file-types = ["templ"]
 roots = ["go.work", "go.mod"]
 comment-token = "//"
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 language-servers = [ "templ" ]
 
 [[grammar]]
@@ -3310,7 +3310,7 @@
 injection-regex = "dbml"
 file-types = ["dbml"]
 comment-token = ";"
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [[grammar]]
 name = "dbml"
@@ -3342,7 +3342,7 @@
 injection-regex = "hoon"
 file-types = ["hoon"]
 comment-token = "::"
-indent = {tab-width = 2, unit = "  "}
+indent = {tab-width = 4, unit = "  "}
 
 [[grammar]]
 name = "hoon"
@@ -3354,7 +3354,7 @@
 file-types = ["conf"]
 comment-token = "#"
 auto-format = true
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [[grammar]]
 name = "hocon"
@@ -3366,7 +3366,7 @@
 injection-regex = "koka"
 file-types = ["kk"]
 comment-token = "//"
-indent = { tab-width = 8, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 language-servers = ["koka"]
 
 [[grammar]]
@@ -3397,7 +3397,7 @@
 injection-regex = "pkl"
 file-types = ["pkl", "pcf"]
 comment-token = "//"
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [[grammar]]
 name = "pkl"
@@ -3410,7 +3410,7 @@
 file-types = ["gradle", "groovy", "jenkinsfile", { glob = "Jenkinsfile" },  { glob = "Jenkinsfile.*" }]
 shebangs = ["groovy"]
 comment-token = "//"
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [[grammar]]
 name = "groovy"
@@ -3454,7 +3454,7 @@
 injection-regex = "ld"
 file-types = ["ld"]
 block-comment-tokens = { start = "/*", end = "*/" }
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [[grammar]]
 name = "ld"
@@ -3550,7 +3550,7 @@
   { start = "/*", end = "*/" },
   { start = "/**", end = "*/" },
 ]
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [language.auto-pairs]
 '"' = '"'
@@ -3571,7 +3571,7 @@
   { glob = "Earthfile" },
 ]
 comment-token = "#"
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 language-servers = ["earthlyls"]
 
 [[grammar]]
@@ -3585,7 +3585,7 @@
 file-types = ["adl"]
 roots = []
 comment-token = "//"
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 
 [language.auto-pairs]
 '"' = '"'
@@ -3680,7 +3680,7 @@
     "vscode-eslint-language-server",
     "ember-language-server",
 ]
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 grammar = "javascript"
 
 [language.auto-pairs]
@@ -3704,7 +3704,7 @@
     "vscode-eslint-language-server",
     "ember-language-server",
 ]
-indent = { tab-width = 2, unit = "  " }
+indent = { tab-width = 4, unit = "  " }
 grammar = "typescript"
 
 [language.auto-pairs]
